schema/AttributeObject.md

```diff
@@ -8,27 +8,39 @@ and in the [Entitlement Object](EntitlementObject.md) to assert the attributes t
 
 Access decisions are made by comparing the attributes all entities have with the attributes a data policy requires.
 
-Attributes that an entity (or actor, or subject) "has" are referred to as "entity entitlements" and are represented by [Entitlement Objects](EntitlementObject.md) 
+Attributes that a single entity (a system actor or subject) "has" are referred to as "entity entitlements" and are represented by [Entity Entitlement Objects](EntitlementObject.md) 
 
-Attributes that entities "need" in order to access data are referred to as "data attributes" and are represented by [Policy Objects](PolicyObject.md)
+Attributes that entities "need" in order to access data are referred to as "data attributes" and are represented by [Data Policy Objects](PolicyObject.md)
 
 The set of all entity entitlements involved in a request are referred to as "claims" and are represented by a [Claims Object](ClaimsObject.md) 
 
-The _attribute_ field must be both globally unique and immutable as it is the reference id for the attribute. 
-All of the other fields are mutable. The attribute string contains three pieces of information - the authority namespace, the attribute name, and the attribute value.
+Attributes themselves are represented as URIs. 
 
-When encrypting, the client determines which attributes an entity must have in order to decrypt the payload and applies those attributes to the file's [Policy Object](PolicyObject.md).
+Given the example attribute URI `https://demo.com/attr/Blob/value/Green`, the named parts of the URI are:
 
-When a decrypt is requested, the KAS checks the [Policy Object](PolicyObject.md) against the [Claims Object](ClaimsObject.md) from the requesting client to 
-ensure the attributes that an entity "has" satisfies those that an entity "needs".
+| Name | Example Value | Description | 
+| ---- | ------------- | ----------- |
+| **Attribute Namespace** | `https://demo.com` | Typically a standard DNS name. It is recommended that the root DNS name of the authoritative owner of the attribute be used as the Attribute Namespace. |
+| **Attribute Name** | `Blob` | Attribute Names are not globally unique. |
+| **Attribute Canonical Name** | `https://demo.com/attr/Blob` | Combination of `Attribute Namespace` and `Attribute Name`, separated by the string `/attr/`. Attribute Canonical Names are the _globally unique_ part of the attribute. |
+| **Attribute Value** | `Green` | Attribute Values are not globally unique. |
+| **Attribute Definition** | `{rule_type: AllOf, valid_values: [Green, Red, Purple]}` | Authorization-relevant metadata (rule type: AllOf/AnyOf/Hierarchy, allowed values, etc) associated with a specific, globally unique `Attribute Canonical Name`. Stored/managed by the authoritative owner of the attribute, separately from data or entity policy. |
```

Comment:
The attribute definition is not included in the example URI - I would suggest maybe putting the attribute instance after the attribute value, and then put the attribute definition?  As a user, when reading, I was confused because I was looking for the definition of everything in the URI first.

Reply:
Agree - I have dithered on how to represent this. I'll move it to the end.